website_portal_restrict_modification/controllers/portal.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2017 - Today Coop IT Easy SC (<http://www.coopiteasy.be>)
+# - Rémy Taymans  <remy@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+import logging
+
+from odoo.addons.portal.controllers.portal import CustomerPortal
+
+_logger = logging.getLogger(__name__)
+
+
+class CustomerPortalRestrictModification(CustomerPortal):
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+        # Class scope is accessible throughout the server even on
+        # odoo instances that do not install this module.
+
+        # Therefore : bring back to instance scope if not already
+        if "MANDATORY_BILLING_FIELDS" not in vars(self):
+            self.MANDATORY_BILLING_FIELDS = (
+                CustomerPortal.MANDATORY_BILLING_FIELDS.copy()
+            )
+        if "OPTIONAL_BILLING_FIELDS" not in vars(self):
+            self.OPTIONAL_BILLING_FIELDS = CustomerPortal.OPTIONAL_BILLING_FIELDS.copy()
+
+        # move name and email to optional
```

Comment:
Why? Isn't it safe to keep these mandatory, even if they're readonly? They seem mandatory on `res.users` _anyway_.

Reply:
I think it's because when they're marked as "readlonly", the fields are not sent back to the controller. Then the controller complains fields are missing.